backend/internal/group/service/groupservice.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package service provides the implementation for group management operations.
+package service
+
+import (
+	"errors"
+
+	"github.com/asgardeo/thunder/internal/group/model"
+	"github.com/asgardeo/thunder/internal/group/store"
+	"github.com/asgardeo/thunder/internal/system/log"
+	"github.com/asgardeo/thunder/internal/system/utils"
+)
+
+// GroupServiceInterface defines the interface for the group service.
+type GroupServiceInterface interface {
```

Comment:
Shall we use [`ServiceError`](https://github.com/asgardeo/thunder/blob/6753309984b727d5fead25e70da98dc232151058/backend/internal/system/error/serviceerror/error.go#L33) struct to return errors from the service layer?

Reply:
Fixed with https://github.com/asgardeo/thunder/compare/73946306ccf2e751a0829d5c8c47baf76a871d36..b8ad696888f32b1b4b8f9e6506e6eb438e2b7ed7